### DISCUSSION TITLE
Feature: cd into the selected directory

### DISCUSSION THREAD

**Discussion by a-glapinski** [PatrickF1/fzf.fish#98] [Ideas]
In default integration available in `fzf`, there is an ability to cd into the selected directory with `ALT-C` keybinding. Is there a possibility to add this functionality to this plugin?

**Comment by PatrickF1** [comment]
Hi @a-glapinski, the find file feature (Ctrl+F). Is intended to replace this (type in cd, hit Ctrl+F, search for the directory, hit enter twice). 
I know it's not as fast and it also shows non-directories, but it does provide the same power. Therefore building a feature just for cd'ing into a sub directory would be a little redundant. And it'll be one more feature for me to build and maintain and for everyone to learn. Hope you understand.

**Comment by finkrer** [comment]
Just stumbled over your project when installing fzf on a new machine, and honestly, this is a deal breaker for me. I could add this myself, I guess, but the same could be said about anything.

It would at least be honest to tell people about this in your comparison with jethrokuan's plugin, because right now, let's admit it, it just tells you this one is plain better. You list the added shortcuts, but not the missing ones.

And, it's not my place to tell you how to do things, but seeing how you have two whole shortcuts for dealing with git, but not such a basic feature as directory switching, makes me think you want this plugin to only have features that you use youself. And that's fair, but again, that's not the impression the readme has left.

**Comment by PatrickF1** [comment]
@finkrer Sorry to hear that. I did not mean to be dishonest, but I guess I could have been clearer. It is 100% possible to replicate the cd with fzf feature, it just takes more work. I even made the folders colored differently so it's easy to distinguish. So I wouldn't call it missing.
FWIW I do use Ctrl+F to cd.

**Comment by PatrickF1** [comment]
For v5, I am going to add keyboard shortcuts for taking action on the selected item, and one of them will be to cd into the folder if it's a folder, and to open the file with $EDITOR if it's not a folder. I haven't had time to work on it though.

**Comment by a-glapinski** [comment]
@PatrickF1 Although directory switching is the feature I use fzf most for, I think I understand your reasoning. I still would like to have this feature, but thanks for creating this plugin, I really like it!

**Comment by finkrer** [comment]
@PatrickF1  Sorry if my words were harsh, I just felt misled after reading such a positive readme.
I will be using the old plugin for now, but your project sounds promising, and I would be happy to return to it later.
As the OP has said, this feature is one of the best uses for fzf. Actions are a step in the right direction, but the thing with productivity, in my opinion, is that common actions like this should take as little effort as possible. The old plugins got it right, which is why people are asking for it right now :)

**Comment by PatrickF1** [comment]
> @PatrickF1 Sorry if my words were harsh, I just felt misled after reading such a positive readme.

Just as a bit of friendly feedback, it did feel harsh. However, I forgive you and I want to empathize with how you felt misled. Let me think about how I can reword it while keeping it succinct. It's a hard problem.

> As the OP has said, this feature is one of the best uses for fzf.

Thanks for giving me this perspective. I heavily rely on `jethrokuan/z`, so I only use my plugin to initially cd into directories for the first time. Do you think this expectation is unreasonable?

**Comment by PatrickF1** [comment]
Hi @a-glapinski @finkrer, there is now something nearly as good as an Alt+C command to cd into the selected dir. See #72

It only requires you to hit enter one extra time after using Ctrl+F.
